### ISSUE TITLE
Broken Linting

### ISSUE THREAD

**Issue by tyjonesAncestry** [mschuchard/linter-terraform-syntax#3]
With version 1.1.2 and linter 2.1.2, I'm no longer getting errors showing up in the terraform code I'm writing. Tried with both validate and plan settings.

Running terraform plan/validate via the CLI, does show errors.

**Comment by mschuchard** [comment]
This is verified working with linter 2.1.2. Can you provide more information about your environment so I can help you troubleshoot?

**Comment by tyjonesAncestry** [comment]
I'm running Atom on macOS Sierra 10.12.4
Terraform Version 0.8.8
I'm running the One Dark theme for both my atom and editor themes.

My project has one folder checked into get with a main.tf and a variables.tf

What else would you like to know?

**Comment by mschuchard** [comment]
Can also verify working with Terraform 0.8.8. Tested with Atom 1.15.0.

Can you provide the files so I can attempt to reproduce?

**Comment by tyjonesAncestry** [comment]
main.tf

module "base_security_group_dedicated_vpc_b" {
  source = "git::GITREPOHERE"
  vpc_id = "${var.dedicated_vpc_b_id}"
}

module "base_public_security_group_dedicated_vpc_b" {
  source = "git::GITREPOHERE"
  vpc_id = "${var.dedicated_vpc_b_id}"
}


variables.tf

##### --DEFAULT VPC IDS--- #####
variable "default_vpc_a_id" {
  default = "vpc-IDHERE"
}


##### --DEFAULT TENANCY VPC IDS--- #####
output "default_vpc_a_id" {
  value = "${var.default_vpc_a_id}"
}

**Comment by mschuchard** [comment]
My package currently outputs the following to the display for these two files:

> Error	Terraform	Non-syntax error in directory: module 'base_security_group_dedicated_vpc_b': unknown variable referenced: 'dedicated_vpc_b_id'. define it with 'variable' blocks.		
Error	Terraform	Non-syntax error in directory: module 'base_public_security_group_dedicated_vpc_b': unknown variable referenced: 'dedicated_vpc_b_id'. define it with 'variable' blocks.		

Is that what you were expecting, or are the variables supposed to be named the same between the two files?

**Comment by tyjonesAncestry** [comment]
That's what I'd be expecting right now, but I don't get that output.

**Comment by tyjonesAncestry** [comment]
I get this:

<img width="855" alt="screen shot 2017-04-06 at 1 16 35 pm" src="https://cloud.githubusercontent.com/assets/26175291/24771559/4c85637e-1acb-11e7-8739-a350662b16a6.png">

**Comment by mschuchard** [comment]
Since the only difference between the test environment and yours is the Darwin versus Linux, I am going to make a guess here. Atom and MacOSX typically have issues preserving your shell environment pathing inside Atom. Since Terraform is installed outside of a package manager, it is probably installed in a non-default location.

Head over to the package settings and place the path to your `terraform` executable inside the `Terraform Executable Path` setting (check in your terminal with `which terraform`). See if that fixes the issue.

**Comment by tyjonesAncestry** [comment]
I just tried that, and still no dice.

I've now tried uninstall and re-installing the packages, and completely removing and reinstalling Atom as well.

<img width="856" alt="screen shot 2017-04-06 at 11 44 04 am" src="https://cloud.githubusercontent.com/assets/26175291/24772822/02c78712-1ad0-11e7-8793-8ca13d9b1a47.png">

**Comment by mschuchard** [comment]
Ok, going to go ahead and escalate.

**Comment by Arcanemagus** [comment]
@tyjonesAncestry Can you open Atom's Developer Tools, go to the Source tab and locate the files for this package and place a breakpoint (click the line number) on [this line](https://github.com/mschuchard/linter-terraform-syntax/blob/v1.1.2/lib/main.js#L46), then trigger a lint on a file you are expecting messages on and check the contents of the `output` variable?

**Comment by tyjonesAncestry** [comment]
The val is set to:

"Error validating: 1 error(s) occurred:

* module 'base_public_security_group_dedicated_vpc_b': unknown variable referenced: 'dedicated_vpc_bb_id'. define it with 'variable' blocks"

However, I never see it in the linter-ui

**Comment by Arcanemagus** [comment]
That looks like a completely different format (and message from what was expected), any idea what's going on @mschuchard?

**Comment by mschuchard** [comment]
@tyjonesAncestry Could you please edit your comment to use Github markdown so that the `output` is formatted verbatim and looks exactly like you are seeing in the editor?

**Comment by mschuchard** [comment]
@tyjonesAncestry Could you also go to the `command palette` and do a `linter: debug` command after linting?

**Comment by tyjonesAncestry** [comment]
```
"Error validating: 1 error(s) occurred:

* module 'base_security_group_default_vpc_a': unknown variable referenced: 'default_vpc_abcs_id'. define it with 'variable' blocks"
```

Linter Debug Info:
```
Platform: darwin
Atom Version: 1.15.0
Linter Version: 2.1.2
All Linter Providers: 
  - Terraform
Matching Linter Providers: 
  - Terraform
Disabled Linter Providers; 
Current File scopes: 
  - *
  - source.terraform
```

**Comment by mschuchard** [comment]
So at this point I can narrow it down to some issue with either MacOSX, Terraform, and/or personal environment/config. Since there are no good MacOSX Vagrant boxes, I will borrow a colleague's Apple laptop on Monday and we will attempt to reproduce this issue.

If we can, this is something with MacOSX and/or Terraform. If we cannot, then this is a personal environment/config issue, and unfortunately I cannot realistically assist with that.

**Comment by mschuchard** [comment]
@tyjonesAncestry Something just occurred to me. By any chance, did you forget to update your settings when you upgraded to linter >= 2? Linter >= 2 no longer displays errors/warnings/info from other files by default, so you have to go to the `linter-ui-default` settings and change them. I made a note in the documentation about this change in Linter >= 2 since it affects three of my packages.

**Comment by tyjonesAncestry** [comment]
That fixed it! Thank you. Guess I must've missed that part.

**Comment by mschuchard** [comment]
Yeah, it wasn't something announced when Linter went to version 2, but the panel no longer displays for all files, and this impacts packages like this that capture issues outside of the current file. I just happened to notice it when i was spec testing my ansible-lint package and noticed the include warning was no longer being displayed. Unfortunately, the best I can do for my three impacted packages is to make a note in the documentation.

Glad this fixed your problem!
